appium-tests/helpers/contactsHelper.js

```diff
@@ -0,0 +1,222 @@
+
+    var result =
+        'var results = document.getElementById("info");\n' +
+        'var obj = new ContactFindOptions();\n' +
+        'obj.filter = ' + preparedOldDisplayName + ';\n' +
+        'obj.multiple = false;\n' +
+
+        'navigator.contacts.find(["displayName", "name"], function(contacts) {\n' +
+        '    if (contacts.length === 0) {\n' +
+        '        results.innerHTML = "No contacts to update.";\n' +
+        '        return;\n' +
+        '    }\n' +
+        '    var contact = contacts[0];\n' +
+        '    contact.displayName = ' + preparedNewDisplayName + ';\n' +
+        '    contact.name = ' + preparedNewName + ';\n' +
+        '    contact.save(function(updated) {\n' +
+        '        results.innerHTML = "Contact updated.";\n' +
+        '    },function(e) {\n' +
+        '        results.innerHTML = "Update failed: error " + e.code;\n' +
+        '    });\n' +
+        '}, function(e) {\n' +
+        '    results.innerHTML = "Search failed: error " + e.code;\n' +
+        '}, obj);';
+
+    return result;
+};
+
+module.exports.getRemoveTestContactsCode = function () {
+    var result =
+        'var results = document.getElementById("info");\n' +
+        'results.innerHTML = "";\n' +
+        'var obj = new ContactFindOptions();\n' +
+        'obj.filter = "DeleteMe";\n' +
+        'obj.multiple = true;\n' +
+        'navigator.contacts.find(["note"], function(contacts) {\n' +
+        '    var removes = [];\n' +
+        '    contacts.forEach(function(contact) {\n' +
+        '        removes.push(contact);\n' +
+        '    });\n' +
+        '    if (removes.length === 0) {\n' +
+        '        results.innerHTML = "No contacts to remove";\n' +
+        '        return;\n' +
+        '    }\n' +
+
+        '   var nextToRemove;\n' +
+        '   if (removes.length > 0) {\n' +
+        '        nextToRemove = removes.shift();\n' +
+        '    }\n' +
+
+        '    function removeNext(item) {\n' +
+        '        if (typeof item === "undefined") {\n' +
+        '            return;\n' +
+        '        }\n' +
+
+        '        if (removes.length > 0) {\n' +
+        '            nextToRemove = removes.shift();\n' +
+        '        } else {\n' +
+        '            nextToRemove = undefined;\n' +
+        '        }\n' +
+
+        '        item.remove(function removeSucceeded() {\n' +
+        '            results.innerHTML += "Removed a contact with ID " + item.id + "<br/>";\n' +
+        '            removeNext(nextToRemove);\n' +
+        '        }, function removeFailed() {\n' +
+        '            results.innerHTML += "Failed to remove a contact with ID " + item.id + "<br/>";\n' +
+        '            removeNext(nextToRemove);\n' +
+        '        });\n' +
+        '    }\n' +
+        '    removeNext(nextToRemove);\n' +
+        '}, function(e) {\n' +
+        '   results.innerHTML = "Search failed: error " + e.code;\n' +
+        '}, obj);';
+
+    return result;
+};
```

Comment:
writing code in strings is a bad pattern indeed. But I don't like the idea to rely on buttons/divs from manual tests. 

I propose the following:
1. Step1. Add helper function to appium which is based on execute, but takes single PARAMETERLESS function as argument repreneting function to be executed on app side. So you will be able to do `return execute(fn.toString())` ti run js code on app side, for example
```
.execute(function() {
    navigator.contacts.create('someParams', winCallback, falseCallback)
})
```
2. Step2. Inject gloabal win and fail functions to be used on app context as a first step before running tests, so you will be able to pass them in all functions, for example `__appiumWinCallback and __appiumFailCallback`

So you will be able to run the following code
```
.execute(function() {
    navigator.contacts.create('someParams', __appiumWinCallback, __appiumFailCallback)
})
```
3. Step2. Inject extra function to return  last execution results (should be based on __appiumWinCallback and __appiumFailCallback implementations), sample usage (appium side)

```
var result = execute('_appiumLastTestResult')
result.win <- raw result from __appiumWinCallback callback (if called)
result.fail <- raw result from __appiumFailCallback fail 

```